src/pivot-table/components/grids/TopGrid.tsx

```diff
@@ -62,11 +62,9 @@ const TopGrid = ({
     height,
   };
 
-  useOnPropsChange(() => {
-    if (topGridRef.current) {
-      topGridRef.current.forEach((list) => list?.resetAfterIndex(0, false));
-    }
-  }, [dataModel, width, height, topDimensionData, topGridRef, headerCellHeight]);
+  useResetListCache(topGridRef, dataModel, topDimensionData);
+
+  useResetListCacheAndRerender(topGridRef, width, height, headerCellHeight);
```

Comment:
Do we need to run both? so resetting twice?

Reply:
I'm looking into just running `useResetListCache` for both data and width/height changes. But both scenarios needs to reset the cache

Reply:
I've removed `useResetListCacheAndRerender`. I'm thinking that `useResetListCache` will run before the react window lists in `LeftGrid` and `TopGrid` renders. So the cache should be reset by then.

Reply:
And some more testing have proven that `useResetListCacheAndRerender` is actually needed. So I will re-add that.

Expand/Collapsing a cell that triggered the vertical scrollbar to appear causes issues when `useResetListCacheAndRerender` is not there